gfx_scene.h

```diff
@@ -1808,9 +1820,10 @@ class GfxSceneInternal
             GltfAnimation &animation_object = gltf_animations_[GetObjectIndex((*it).second)];
             for(size_t j = 0; j < gltf_scene.nodes.size(); ++j)
             {
-                std::map<int32_t, uint64_t>::const_iterator const it2 = animated_nodes.find(gltf_scene.nodes[j]);
-                if(it2 != animated_nodes.end())
-                    animation_object.nodes_.push_back((*it2).second);
+                std::map<int32_t, std::pair<size_t, uint64_t>>::const_iterator const it2 = animated_nodes.find(gltf_scene.nodes[j]);
+                GFX_ASSERT(it2 == animated_nodes.end() || (*it2).second.first < gltf_model.animations.size());
+                if(it2 != animated_nodes.end() && (*it2).second.first == i)
```

Comment:
We discard animated nodes belonging to another animation.

Reply:
If we don't do this, each animated nodes will be updated by each animation. If we only play one animation at a time, it works but when a few or all animations are played, it can lead to weird results.